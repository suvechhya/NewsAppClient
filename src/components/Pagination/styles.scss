@import '../../styles/main.scss';

.pagination{
    &--btn{
        background-color: $secondary-text;
        color: $primary-bg;
        border: none;
        outline: none;

        &.active{
            background-color: $primary-bg;
            color: $secondary-text;
        }
    }
}