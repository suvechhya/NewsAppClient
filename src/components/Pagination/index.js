import React, { useEffect, useState } from 'react';
import './styles.scss';
import clsx from 'clsx';

const Header = ({limit, selected, parentCallback}) => {
    const [items, setItems] = useState([]);
    useEffect(() => {
        if(limit && limit>0) {
            const tempItems = [];
            for(let i = 1; i<=limit; i++) {
                tempItems.push({
                    label: i,
                    value: i
                })
            }
            setItems([...tempItems]);
        }
        
    }, [limit]);

  return (
    <nav className="pagination d-flex justify-content-center">
        {items && items.length > 0 && items.map((i) => (
            <button id={`pagebtn${i.value}`} key={`pagebtn${i.value}`} className={clsx('pagination--btn', {'active': selected === i.value})} onClick={() => parentCallback({newPage: i.value})}>{i.label}</button>
        ))}
    </nav>
  );
}

export default Header;
