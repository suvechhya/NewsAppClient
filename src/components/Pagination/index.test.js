import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json'
import Pagination from './index';

describe('Pagination', () => {
    it('should render correctly with given props', () => {
        const blankFn = () => {};
        const component = shallow(<Pagination limit={4} selected={1} parentCallback={blankFn} />);
        expect(component).toMatchSnapshot();
    });

    it("should render correctly each time", () => {
        const blankFn = () => {};
        const tree = shallow(<Pagination limit={4} selected={1} parentCallback={blankFn} />);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('should be possible to activate page buttons with Spacebar', () => {
        const blankFn = () => {};
        const component = mount(<Pagination limit={4} selected={1} parentCallback={blankFn} />);
        component
            .find('nav')
            .childAt(1)
            .simulate('keydown', { keyCode: 32 });
        expect(component).toMatchSnapshot();
        component.unmount();
    });

    const clickFn = jest.fn();
    it('props function should be callable', () => {
        const component = mount(<Pagination limit={4} selected={1} parentCallback={clickFn} />);
        component
            .find('nav')
            .childAt(1)
            .simulate('click');
        expect(clickFn).toHaveBeenCalled();
    });
});
