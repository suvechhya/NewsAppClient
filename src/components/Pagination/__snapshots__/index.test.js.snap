// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should be possible to activate page buttons with Spacebar 1`] = `ReactWrapper {}`;

exports[`Pagination should render correctly each time 1`] = `
<nav
  className="pagination d-flex justify-content-center"
/>
`;

exports[`Pagination should render correctly with given props 1`] = `ShallowWrapper {}`;
